credentialdigger/client.py

```diff
@@ -254,22 +250,18 @@ def get_repo(self, query, repo_url):
         -------
         dict
             A repository (an empty dictionary if the url does not exist)
+
+        Raises
+        ------
+            TypeError
+                If any of the required arguments is missing
         """
         cursor = self.db.cursor()
-        try:
-            cursor.execute(query, (repo_url,))
-            result = cursor.fetchone()
-            if result:
-                return dict(Repo(*result)._asdict())
-            else:
-                return {}
-        except (TypeError, IndexError):
-            """ A TypeError is raised if any of the required arguments is
-            missing. """
-            self.db.rollback()
-            return {}
-        except self.Error:
-            self.db.rollback()
+        cursor.execute(query, (repo_url,))
+        result = cursor.fetchone()
+        if result:
+            return dict(Repo(*result)._asdict())
+        else:
             return {}
 
     def get_rules(self, category_query, category=None):
```

Comment:
why is `category_query` positional and not optional?

Reply:
Uhm I didn't touch that, it's the same in the main branch as well, but I'll go ahead and make it optional because it has no purpose in being mandatory.

https://github.com/SAP/credential-digger/blob/dd58f5eaa6f316f50589218c7a6273555eeb18ab/credentialdigger/client.py#L270-L290